internal/provider/destination_resource.go

```diff
@@ -230,3 +231,39 @@ func (r *DestinationResource[T]) Update(ctx context.Context, req resource.Update
 func (r *DestinationResource[T]) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
 	resp.Schema = r.schema
 }
+
+func (r *DestinationResource[T]) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	parts := strings.Split(req.ID, "/")
```

Comment:
just to confirm my understanding - it is the export process that puts the IDs into this format?

Reply:
TF only allows a single string value to be used to identify every resource but our API requires the pipeline id and the component id. Mashing the two ids with a `/` separator is a fairly standard way that providers work around this restriction.

Reply:
yeah that I get, but there's nothing _in the provider_ that is generating those IDs correct? we're just validating them now in the `ImportState` functions, and generating those IDs when outputting resource definitions from the exporter?

Reply:
> but there's nothing in the provider that is generating those IDs correct?

Correct. This change just takes the user input, attempts to fetch the component details from the API and stores those values if the API call succeeds.

> generating those IDs when outputting resource definitions from the exporter

Not yet. The export tool needs to be updated to generate a list of command like:
```
terraform import 689f9646-83f2-11ee-b962-0242ac120002/689f9ad8-83f2-11ee-b962-0242ac120002
```
in a script. Assuming the user can find the UUID values for the pipeline and the components, they could run each command themselves (this will likely never happen if we don't provide a script).

Reply:
got it, thanks for the additional context as per usual